frontend/src/services/ApiService.tsx

```diff
@@ -0,0 +1,44 @@
+import ServiceProvider from "./ServiceProvider";
+import ProvidedServices from "./ProvidedServices";
+import axios from "axios";
+import { useEffect, useRef, useState } from "react";
+
+export interface IApiService {
+  login(user: string, pass: string): void;
+}
+
+const ApiServiceContext = ServiceProvider.createContext(
```

Comment:
I see some value in creating a User Service. Since the user information might be accessed at multiple points in the application, it would be useful to have access to that as state at a global level.